tools/ci_build/build.py

```diff
@@ -1598,6 +1598,7 @@ def run_onnxruntime_tests(args, source_dir, ctest_path, build_dir, configs):
                 cwd = os.path.join(cwd, config)
 
             run_subprocess([sys.executable, 'onnxruntime_test_python.py'], cwd=cwd, dll_path=dll_path)
+            run_subprocess([sys.executable, 'orttraining_test_gradient_graph.py'], cwd=cwd, dll_path=dll_path)
```

Comment:
Let's add a comment here explaining what this test is.
Let's also move it under the method: `run_training_python_frontend_tests`